daml-lf/interpreter/src/main/scala/com/digitalasset/daml/lf/speedy/Compiler.scala

```diff
@@ -297,6 +305,20 @@ private[lf] final class Compiler(
   private[this] def compileCommands(cmds: ImmArray[Command]): t.SExpr =
     pipeline(translateCommands(Env.Empty, cmds))
 
+  private[this] def compileWithContractDisclosures(
+      sexpr: t.SExpr,
+      disclosures: ImmArray[DisclosedContract],
+  ): t.SExpr = {
+    val disclosureLambda = pipeline(
+      translateContractDisclosureLambda(Env.Empty, disclosures)
+    )
+
+    t.SELet1(
```

Comment:
If `sexpr` is in ANF form, then this resulting expression should also be in ANF form